sql: implement pg_lsn operations

otan opened the pull request:
## sql: add pg_lsn +- decimal operations

Release note (sql change): Add the ability to add numeric values to
LSNs, or sub a decimal value from a LSN.

## tree: implement lsn - lsn operation

Release note (sql change): Implement the `pg_lsn - pg_lsn = decimal`
builtin, which subtracts 2 LSNs to return a decimal.

----

Informs #105130

otan committed:
sql: add pg_lsn +- decimal operations

Release note (sql change): Add the ability to add numeric values to
LSNs, or sub a decimal value from a LSN.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/105326)
<!-- Reviewable:end -->

otan committed:
tree: implement lsn - lsn operation

Release note (sql change): Implement the `pg_lsn - pg_lsn = decimal`
builtin, which subtracts 2 LSNs to return a decimal.

ZhouXing19 reviewed (approved):
> Reviewed 7 of 7 files at r1, 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/105326)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=zhouxing19
